pkg/collector/osmlogs_collector.go

```diff
@@ -34,34 +34,20 @@ func (collector *OsmLogsCollector) Collect() error {
 	// can define resources to query in deployment.yaml and iterate through the commands+resources needed and create multiple files
 	// see kubeobject collector for example
 
-	// * Get osm resources as table
-	allResourcesFile := filepath.Join(rootPath, "allResources")
-	output, err := utils.RunCommandOnContainer("kubectl", "get", "all", "--all-namespaces", "--selector", "app.kubernetes.io/name=openservicemesh.io")
-	if err != nil {
-		return err
+	// ** Get ground truth on OSM resources **
+	var groundTruthMap = map[string][]string{
+		"allResourcesTable":               []string{"get", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=wide"},
+		"allResourcesConfigs":             []string{"get", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
+		"mutatingWebhookConfigurations":   []string{"get", "MutatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
+		"validatingWebhookConfigurations": []string{"get", "ValidatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json"},
 	}
 
-	err = utils.WriteToFile(allResourcesFile, output)
-	if err != nil {
-		return err
-	}
-
-	collector.AddToCollectorFiles(allResourcesFile)
-
-	// * Get osm resource configs
-	allResourceConfigsFile := filepath.Join(rootPath, "allResourceConfigs")
-	output, err = utils.RunCommandOnContainer("kubectl", "get", "all", "--all-namespaces", "--selector", "app.kubernetes.io/name=openservicemesh.io", "-o", "json")
-	if err != nil {
-		return err
-	}
-
-	err = utils.WriteToFile(allResourceConfigsFile, output)
-	if err != nil {
-		return err
+	for fileName, kubeCmd := range groundTruthMap {
+		if err = collectResourceToFile(collector, rootPath, fileName, kubeCmd); err != nil {
+			return err
```

Comment:
I wonder if this should log an error and `continue` instead of completely stopping the execution of the file 🤔